File: rest-pgclient/src/main/java/com/example/Post.java
package com.example;

import java.time.LocalDateTime;
import java.util.UUID;

public record Post(
        UUID id,
        String title,
        String content,
        LocalDateTime createdAt) {

    public static Post of(String title, String content) {
        return new Post(null, title, content, null);
    }
}

File: mongodb-kotlin/src/main/kotlin/com/example/Post.kt
package com.example

import io.quarkus.mongodb.panache.common.MongoEntity
import org.bson.codecs.pojo.annotations.BsonId
import org.bson.types.ObjectId
import java.time.LocalDateTime


@MongoEntity
data class Post(
    @BsonId
    var id: ObjectId? = null,// used by MongoDB for the _id field var
    var title: String? = null,
    var body: String? = null,
    var createdAt: LocalDateTime? = LocalDateTime.now()
) {
   // constructor() : this(null, null, null, null)
}
